### DISCUSSION TITLE
double max to string to double result in infinit using IMX6 CPU and QNX operating system

### DISCUSSION THREAD

**Discussion by SchneebergerF** [nlohmann/json#3251] [Q&A] [answered]
Using the following code

```cpp
#include <limits> // std::numeric_limits
#include <sstream>
#include <iostream>

void foo()
{
std::sstream ss;
double val = std::numeric_limits<double>::max();
ss.precision(std::numeric_limits<double>::max_digits10);
ss << val;
std::cout << ss.str() << std::endl;
double val2 = std::strtod(ss.str().c_str(), nullptr);
std::cout << val2 << std::endl;
}
```

the results are machine dependent, i.e.
Welcome to the Momentics IDE
QNX noname 7.0.4 2019/08/02-15:56:44EDT **x86pc x86_64**
1.7976931348623157e+308
1.79769e+308

Welcome to the Momentics IDE
Linux ubuntu 4.15.0-66-generic #75-Ubuntu SMP Tue Oct 1 05:24:09 UTC 2019 **x86_64 x86_64 x86_64 GNU/Linux**
1.7976931348623157e+308
1.79769e+308

Welcome to the Momentics IDE
QNX localhost 7.0.4 2019/08/02-15:50:58EDT **i.MX6_QuadorDual_Sabre-Smart_Board armle**
1.7976931348623157e+308
inf 

As you can see, the I.MX6 CPU does not handle the max() value correctly.
According to QNX support, the results comes from the ss.precision(std::numeric_limits<double>::max_digits10);
Removing this line does not return inf while executing the code on the I.MX6 CPU,

The code sample has been derived from the function dump_float() present in the JSON library. Refer to file serializer.hpp (line 794 - V3.9.1)

Is there anything that can be done for this specific case ?
Thanks

**Comment by nlohmann** [comment, marked as answer]
This sounds like a bug in the implementation of `std::strtod`, right?

**Reply by nlohmann** [reply]
@SchneebergerF 
Do you need further assistance with this issue?
